### PR TITLE
Removed tags from s3 plural resource initialization

### LINKED ISSUE #819: Poorly Performing API Call in aws_s3_buckets Resource
The aws_s3_buckets resource makes an additional call to get the tags of every bucket returned by the list_buckets method. This leads to poor performance when the list_buckets API returns many results.

# Describe the problem
I would expect that the tags are not fetched in the plural resource, since the singular resource fetches the tags anyway.
image

## Possible Solution
Remove the tags field from the plural resource. Allow the end user to fetch tags with the singular resource if they need the tags.

### PR DESCRIPTION
**Pull request by jnikles1**
Signed-off-by: Justin Nikles <justin.nikles@sap.com>

### Description

Removed the tags attribute from the plural resource, as this adds unnecessary overhead when there are many S3 buckets, due to the fact that it makes 1 API call per bucket to get the tags. Tags can be fetched in the singular resource.

### Issues Resolved

closes #819 

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- N/A New functionality includes integration tests/controls
- N/A New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by jnikles1** [comment]
@sathish-progress - please let me know if you have any suggestions regarding lazy loading as mentioned in #819. I investigated lazy loading, but it appears that the filter table lazy loading cannot be used as it does not allow you to access instance methods or instance variables. The AWS client cannot be accessed in a class method, so it appears this is not possible. That is why I've completely removed the tags attribute

File: libraries/aws_s3_buckets.rb [hunk 1]
```diff
@@ -15,7 +15,6 @@ class AwsS3Buckets < AwsResourceBase
 
   FilterTable.create
              .register_column(:bucket_names, field: :bucket_name)
-             .register_column(:tags,         field: :tags)
```
Comment (sathish-progress) [inline libraries/aws_s3_buckets.rb:18]:
```suggestion
            .register_column(:tags, field: :tags, lazy: ->() { |row,_c,_t| row[:tags] = fetch_tags(row[:bucket_name]) })
```

**Comment by sathish-progress** [comment]
> @sathish-progress - please let me know if you have any suggestions regarding lazy loading as mentioned in #819. I investigated lazy loading, but it appears that the filter table lazy loading cannot be used as it does not allow you to access instance methods or instance variables. The AWS client cannot be accessed in a class method, so it appears this is not possible. That is why I've completely removed the tags attribute

@jnikles1 , could you try this out https://github.com/inspec/inspec-aws/pull/821/files#r789662270 ?

File: libraries/aws_s3_buckets.rb (hunk 1 above)
Comment (jnikles1) [reply libraries/aws_s3_buckets.rb:18]: @sathish-progress - I've tried this already. But the syntax gives this error 
```
/var/lib/gems/2.7.0/gems/inspec-core-4.52.9/lib/inspec/profile_context.rb:170:in `instance_eval': libraries/aws_s3_buckets.rb:18: syntax error, unexpected '|' (SyntaxError)
...s, field: :tags, lazy: ->() { |row,_c,_t| row[:tags] = fetch...
...                              ^
libraries/aws_s3_buckets.rb:18: syntax error, unexpected '|', expecting '='
...:tags, lazy: ->() { |row,_c,_t| row[:tags] = fetch_tags(row[...
...
```

Additionally, it seems that instance methods are not accessible in the lazy load of a filter table

**Comment by jnikles1** [comment]
Hi @sathish-progress - please see my inline comment above

**Comment by sathish-progress** [comment]
@jnikles1 , Hey let me know how this looks https://github.com/inspec/inspec-aws/pull/834

**Comment by jnikles1** [comment]
Hi Sathish - that looks good. I will try it out on one of our large accounts now.

**Comment by sathish-progress** [comment]
@jnikles1 , closing this .. since it is resolved as part of #834

**Pull request closed without merging**